Flaky test "SubscribeTest.test_subscribe_push_bound"

### Observed behavior

Pytest `SubscribeTest.test_subscribe_push_bound` occasionally fails as illustrated in [this failed test run](https://github.com/nats-io/nats.py/actions/runs/14792445728/job/41633674988#step:5:233):
```
=================================== FAILURES ===================================
___________________ SubscribeTest.test_subscribe_push_bound ____________________
[...]
>       assert sub2.pending_msgs == 9
E       assert 2 == 9
E        +  where 2 = <nats.js.client.JetStreamContext.PushSubscription object at 0x7f80cb50d5b0>.pending_msgs

tests/test_js.py:1644: AssertionError
```

There is a race condition between the `asyncio` task inserting messages into the pending message queue from method `nats.aio.client.Client._process_msg()` and the task running the test function's `assert sub2.pending_msgs == 9`, causing the latter to inspect the length of the pending message queue before the former task has had time to insert all the pending messages arriving from the NATS server.

### Expected behavior

Test `SubscribeTest.test_subscribe_push_bound` should not be flaky and always succeed.

### Server and client version

nats-server: v2.11.2-RC.1
nats.py: v2.10.0

### Host environment

_No response_

### Steps to reproduce

Run the test for while and observe that it eventually fails:
```
while pytest -k test_subscribe_push_bound; do :; done
```

--- a/tests/test_js.py
+++ b/tests/test_js.py
@@ -1615,11 +1615,11 @@ class SubscribeTest(SingleJetStreamServerTestCase):
 
         # Create a sync subscriber now.
         sub2 = await js.subscribe("pbound", durable="two")
-        msg = await sub2.next_msg()
-        assert msg.data == b"Hello World 0"
-        assert msg.metadata.sequence.stream == 1
-        assert msg.metadata.sequence.consumer == 1
-        assert sub2.pending_msgs == 9
+        for i in range(10):
+            msg = await sub2.next_msg()
+            assert msg.data == f"Hello World {i}".encode()
+            assert msg.metadata.sequence.stream == i + 1
+            assert msg.metadata.sequence.consumer == i + 1
 
         await nc.close()
 

